### PR TITLE
add entityAllowedToRespond and entitiesRequiredToRespond to card template (#2092)

### PR DESCRIPTION
**Pull request by olivierPigeon-RTE**
In Release note, feature section : 
#2092 : Add entityAllowedToRespond and entitiesRequiredToRespond to card template

Signed-off-by: olivierPigeon-RTE <olivier.pigeon@rte-france.com>

### CODE REVIEW

File: services/businessconfig/src/main/java/org/opfab/businessconfig/model/UserCardData.java [hunk 1]
```diff
@@ -47,4 +47,9 @@
     private Boolean recipientVisible;
     @JsonInclude(JsonInclude.Include.NON_EMPTY)
     private List<Recipient> recipientList;
+    @JsonInclude(JsonInclude.Include.NON_EMPTY)
+    private List<Recipient> entitiesAllowedToRespond;
+    @JsonInclude(JsonInclude.Include.NON_EMPTY)
+    private List<Recipient> entitiesRequiredToRespond;
+
```
Comment (freddidierRTE) [inline services/businessconfig/src/main/java/org/opfab/businessconfig/model/UserCardData.java:54]: You do not need that

File: services/businessconfig/src/main/modeling/swagger.yaml [hunk 2]
```diff
@@ -549,6 +549,14 @@ definitions:
...
         items:
           $ref: '#/definitions/Recipient'
+      entitiesAllowedToRespond:
+        type: array
+        items:
+          $ref: '#/definitions/Recipient'
+      entitiesRequiredToRespond:
+        type: array
+        items:
+          $ref: '#/definitions/Recipient'
```
Comment (freddidierRTE) [inline services/businessconfig/src/main/modeling/swagger.yaml:559]: You do not need that

File: services/cards-consultation/src/main/java/org/opfab/cards/consultation/model/ArchivedCardConsultationData.java [hunk 3]
```diff
@@ -81,9 +81,9 @@
     private List<String> groupRecipients;
     @Singular
     private List<String> entityRecipients;
-    @Singular("entityAllowedToRespond")
+    @Singular("entitiesAllowedToRespond")
     private List<String> entitiesAllowedToRespond;
-    @Singular("entityRequiredToRespond")
+    @Singular("entitiesRequiredToRespond")
```
Comment (freddidierRTE) [inline services/cards-consultation/src/main/java/org/opfab/cards/consultation/model/ArchivedCardConsultationData.java:86]: Let as it is , it must be in the singular and not plural  ,have a look to this article : 

 https://www.baeldung.com/lombok-builder-singular

File: services/cards-consultation/src/main/java/org/opfab/cards/consultation/model/CardConsultationData.java [hunk 4]
```diff
@@ -80,9 +80,9 @@
     private List<String> groupRecipients;
     @Singular
     private List<String> entityRecipients;
-    @Singular("entityAllowedToRespond")
+    @Singular("entitiesAllowedToRespond")
     private List<String> entitiesAllowedToRespond;
-    @Singular("entityRequiredToRespond")
+    @Singular("entitiesRequiredToRespond")
```
Comment (freddidierRTE) [inline services/cards-consultation/src/main/java/org/opfab/cards/consultation/model/CardConsultationData.java:85]: Same comments as previous one

File: services/cards-publication/src/main/java/org/opfab/cards/publication/model/ArchivedCardPublicationData.java [hunk 5]
```diff
@@ -88,10 +88,10 @@
     
     private List<String> externalRecipients;
     
-    @Singular("entityAllowedToRespond")
```
Comment (freddidierRTE) [inline services/cards-publication/src/main/java/org/opfab/cards/publication/model/ArchivedCardPublicationData.java:91]: Same comments as previous one

File: services/cards-publication/src/main/java/org/opfab/cards/publication/model/CardPublicationData.java [hunk 6]
```diff
@@ -102,9 +102,9 @@
     @JsonInclude(JsonInclude.Include.NON_EMPTY)
     @Indexed
     private List<String> groupRecipients;
-    @Singular("entityAllowedToRespond")
+    @Singular("entitiesAllowedToRespond")
     private List<String> entitiesAllowedToRespond;
-    @Singular("entityRequiredToRespond")
+    @Singular("entitiesRequiredToRespond")
```
Comment (freddidierRTE) [inline services/cards-publication/src/main/java/org/opfab/cards/publication/model/CardPublicationData.java:107]: Same comments as previous one

File: ui/main/src/app/modules/usercard/usercard.component.ts [hunk 7]
```diff
@@ -109,6 +109,8 @@ export class UserCardComponent implements OnDestroy, OnInit {
     endDateVisible = true ;
     lttdVisible = true ;
     recipientVisible = true; // if recipientVisible == false, then selectedRecipients = recipientList
+    entitiesAllowedToRespond;
+    entitiesRequiredToRespond;
```
Comment (freddidierRTE) [inline ui/main/src/app/modules/usercard/usercard.component.ts:113]: It seems you do not used these variables ?

**Review by freddidierRTE — CHANGES_REQUESTED** [review]

File: ui/main/src/app/modules/usercard/usercard.component.ts (hunk 7 above)
Comment (olivierPigeon-RTE) [reply ui/main/src/app/modules/usercard/usercard.component.ts:113]: yes, they are used below, in lines 555-560

Comment (freddidierRTE) [reply ui/main/src/app/modules/usercard/usercard.component.ts:113]: in 555-560 it s local variables so not these ones

Comment (olivierPigeon-RTE) [reply ui/main/src/app/modules/usercard/usercard.component.ts]: fixed

**Comment by freddidierRTE** [comment]
https://opfab.github.io/documentation/current/reference_doc/#_method_getspecificcardinformation

**Pull request merged**